### ISSUE TITLE
Migrator for "Mixed Declarations" deprecation

### ISSUE THREAD

**Issue by lfloyd117** [sass/migrator#261]
Following [this comment](https://github.com/sass/sass/issues/3926#issuecomment-2344828398), I wanted to open a feature request here for a tool to assist in properly migrating Mixed Declaration deprecations as I did not see one open. 

https://sass-lang.com/documentation/breaking-changes/mixed-decls/

**Comment by nex3** [comment]
Note that for this migration in particular, it's totally reasonable to look over your existing violations, determine that none of them are actually going to change visual styling under the new behavior, and just silence the warning until the breaking change happens.

**Comment by lfloyd117** [comment]
Is there a way to enable an experimental flag for the new behavior to test? We have thousands of cases, so one-by-one is a tad tricky

**Comment by nex3** [comment]
There's not—we generally try to avoid having flags that change the behavior of the core compiler because it makes it much harder to share styles across applications. That said, we could potentially add a `--future-changes` flag analogous to `--future-deprecations` specifically for CSS compatibility changes that are going to come in a minor version, since libraries need to be flexible about them anyway.

**Comment by lfloyd117** [comment]
> There's not—we generally try to avoid having flags that change the behavior of the core compiler because it makes it much harder to share styles across applications. That said, we could potentially add a `--future-changes` flag analogous to `--future-deprecations` specifically for CSS compatibility changes that are going to come in a minor version, since libraries need to be flexible about them anyway.

That sounds like it'd be incredibly helpful. Should that be a new feature request issue?

**Comment by nex3** [comment]
Yes, feel free to file it in the sass/sass repo.

**Comment by nex3** [comment]
There are two reasons we don't want to make a migrator for this.

1. Philosophically, it's not clear that doing the no-op change is actually correct. The reason the CSS spec has different behavior than Sass here is because the old Sass behavior is unintuitive; it applies styles in a different order than they're written in the document. It's possible, even potentially likely, that this is a bug in authors' code, especially in places where styles are explicitly written rather than coming from mixins.

2. Technically, if the styles _are_ coming from mixins, it's not clear that we _can_ do an automated fix. If a rule contains mixins that each add top-level styles as well as nested styles, there's not a good way to fix that beyond a substantial refactor that's outside the scope of a migration tool.

In other words: the places we could feasibly migrate are the same places where it's least likely to be desirable to migrate, so building a migrator just doesn't make a lot of sense.

**Comment by Vinnl** [comment]
Gotcha, thanks for the elaborate clarification!
